### PR TITLE
added dsa problems under leetcode folder

### PR DESCRIPTION
**Pull request by kaustav1810**

### CODE REVIEW

**Comment by kaustav1810** [comment]
@deepsubha akhon e merge korish nah

**Comment by deepsubha** [comment]
@kaustav1810 ready for merge hole bolis. merge koris na ami dekhbo ekbar problem gulo

File: LeetCode/lexicographically_smallest_palindrome.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class Solution {
+    public String makeSmallestPalindrome(String s) {
+        
+        int n = s.length();
+        
+        if(n==1) return s;
+        
+        int l=0,r = n-1;
+        
+        StringBuilder str = new StringBuilder(s);
```
Comment (deepsubha) [inline LeetCode/lexicographically_smallest_palindrome.java]: @kaustav1810 nice 🚀.
middle thekeo du dike traverse kora jeto same e hoto ekta variable kom hoto just

File: LeetCode/minimun_string_length_after_removing_substrings.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// problem link: https://leetcode.com/problems/minimum-string-length-after-removing-substrings/
```
Comment (deepsubha) [inline LeetCode/minimun_string_length_after_removing_substrings.java]: link gulo hyperlink kore dis nahole copy kore new tab khule paste korte hoche @kaustav1810

File: LeetCode/strong_students_in_school.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// problem link: https://practice.geeksforgeeks.org/contest/gfg-weekly-coding-contest-102/problems/#
```
Comment (deepsubha) [inline LeetCode/strong_students_in_school.java]: Hyperlink @kaustav1810

File: LeetCode/minimun_string_length_after_removing_substrings.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * 
+ * Complexity:
+ * --------------
+ * T.C: O(n)
+ * S.C: O(n)
+ */
+
+import java.util.Stack;
+
+class Solution {
```
Comment (deepsubha) [inline LeetCode/minimun_string_length_after_removing_substrings.java]: @kaustav1810 tui koto kichu korechis eta to ekdom easy problem 😠 . bcz AB and CD duto hardcoded string diyache

public int minLength(String s) {
        if(s == null) return 0;
        if(s.length() == 1) return 1;
        while (s.contains("AB") || s.contains("CD")){
            s = s.replace("AB","");
            s = s.replace("CD","");
        }
        return s.length();
    }

File: LeetCode/strong_students_in_school.java (hunk 3 above)
Comment (deepsubha) [reply LeetCode/strong_students_in_school.java]: acha eta GFG er chilo.. bah! nice 🚀

**Review by deepsubha — APPROVED** [review]
cool😎 .. merge kore de draft ache ekhono PR ta .

File: LeetCode/minimun_string_length_after_removing_substrings.java (hunk 4 above)
Comment (kaustav1810) [reply LeetCode/minimun_string_length_after_removing_substrings.java]: ishh....eta dekhe shotti lojja laglo nijer upor. Java string er je contains() method ache mne chilo nah....ei problem tah dicchi nah tahole....eto easy

File: LeetCode/minimun_string_length_after_removing_substrings.java (hunk 2 above)
Comment (kaustav1810) [reply LeetCode/minimun_string_length_after_removing_substrings.java]: Haa okay kore dicchi

File: LeetCode/lexicographically_smallest_palindrome.java (hunk 1 above)
Comment (kaustav1810) [reply LeetCode/lexicographically_smallest_palindrome.java]: l=0,r=(n-i-1) kore toh...?

**Review by deepsubha — APPROVED** [review]

**Comment by kaustav1810** [comment]
[] () diye problem link tah hyperlink korlam, jani nah kaaj korbe kina. Error aschilo bole '.java' file extension o soriye dilam. Tui ekbar dekhe merge kore dish. Ei PR tay r notun problem add korbo nah.

File: LeetCode/lexicographically_smallest_palindrome.java (hunk 1 above)
Comment (deepsubha) [reply LeetCode/lexicographically_smallest_palindrome.java]: middle index.  ====>.    m = n / 2
m+1 / m-1 korle ager or next index ta pabi. unmatch pele return false ...puro loop traverse hole last e return true..

File: LeetCode/minimun_string_length_after_removing_substrings.java (hunk 4 above)
Comment (deepsubha) [reply LeetCode/minimun_string_length_after_removing_substrings.java]: re dibi na keno ekta stack diye apply korechis khoti ki.. similar solutionating tor onno kono problem e kaj e lege jabe similar approach

**Comment by deepsubha** [comment]
@kaustav1810 merge kore dilam eta. hyperlink na hole ki kora jbe ar. []() e hyperlink hobe na only .md file e hobe mone hoi.. leave it ..

**Pull request merged**